dandi/upload.py

```diff
@@ -314,9 +317,30 @@ def upload_agg(*ignored: Any) -> str:
                     else:
                         rec[tuple(rec_fields[1:])] = process_path(dfile)
                 except ValueError as exc:
-                    rec.update(skip_file(exc))
+                    rec.update(error_file(exc))
                 out(rec)
 
+        if not validate_ok:
+            lgr.warning(
+                "One or more assets failed validation.  Consult the logfile for"
+                " details."
+            )
+        if not ok:
+            try:
+                import etelemetry
+
+                latest_version = etelemetry.get_project("dandi/dandi-cli")["version"]
+            except Exception:
+                pass
+            else:
+                if Version(latest_version) > Version(__version__):
+                    lgr.warning(
+                        "Upload failed, and you are not using the latest"
+                        " version of dandi.  Suggest upgrading dandi and trying"
+                        " again."
+                    )
+            sys.exit(1)
```

Comment:
could we `exit(1)` at `cmd_upload.py` level, not here, which is just a "high level" Python API... I think we do not even need to exit(1) explicitly and our `click` wrapper would do it.  Let's just (re)raise `UploadError` exception here?

Reply:
The `UploadError` messages are all rather terse and don't exactly make for user-friendly exit messages.

Reply:
Agree, but for now I think it would suffice.  It would still be better than throwing SystemExit into user while calling `upload` function somewhere within IPython notebook.

Reply:
Done in b6b9195.